subprojects/snapshots/src/main/java/org/gradle/internal/vfs/VirtualFileSystem.java

```diff
@@ -36,7 +38,13 @@ public interface VirtualFileSystem {
     /**
      * Adds the information of the snapshot at the absolute path to the VFS.
      */
-    void store(String absolutePath, FileSystemLocationSnapshot snapshot);
+    FileSystemLocationSnapshot store(String absolutePath, Supplier<FileSystemLocationSnapshot> snapshotSupplier);
+
+    FileSystemLocationSnapshot store(String baseLocation, StoringAction storingAction);
```

Comment:
Let's explain how these methods now work in Javadoc.

Reply:
Done.